platforms/nuttx/src/px4/stm/stm32_common/dshot/dshot.c

```diff
@@ -0,0 +1,440 @@
+		rS6CR(timer) |= DMA_SCR_TCIE | DMA_SCR_HTIE | DMA_SCR_TEIE | DMA_SCR_DMEIE;
+		rS6PAR(timer)  =  io_timers[timer].base + STM32_GTIM_DMAR_OFFSET;
+		rS6M0AR(timer) = (uint32_t)(dshot_burst_buffer[timer]);
+		rS6FCR(timer) &= 0x0;  /* Disable FIFO */
+		break;
+
+	case DShot_Stream7:
+		rS7CR(timer) |= DMA_SCR_CHSEL(io_timers[timer].dshot.channel);
+		rS7CR(timer) |= DMA_SCR_PRIHI | DMA_SCR_MSIZE_32BITS | DMA_SCR_PSIZE_32BITS | DMA_SCR_MINC | DMA_SCR_DIR_M2P;
+		rS7CR(timer) |= DMA_SCR_TCIE | DMA_SCR_HTIE | DMA_SCR_TEIE | DMA_SCR_DMEIE;
+		rS7PAR(timer)  =  io_timers[timer].base + STM32_GTIM_DMAR_OFFSET;
+		rS7M0AR(timer) = (uint32_t)(dshot_burst_buffer[timer]);
+		rS7FCR(timer) &= 0x0;  /* Disable FIFO */
+		break;
+
+	default:
+		ret_val = ERROR;
+		break;
+	}
+
+	return ret_val;
+}
+
+void up_dshot_trigger(void)
+{
+	uint8_t first_motor = 0;
+
+	for (uint8_t timer = 0; (timer < DSHOT_TIMERS); timer++) {
+
+		if (true == dshot_handler[timer].init) {
+
+			uint32_t dma_int_streamx_mask;
+			volatile uint32_t *rSxNDTR;
+			volatile uint32_t *rSxCR;
+
+			switch (io_timers[timer].dshot.stream) {
+			case DShot_Stream0:
+				dma_int_streamx_mask = DMA_INT_STREAM0_MASK;
+				rSxNDTR = &rS0NDTR(timer);
+				rSxCR = &rS0CR(timer);
+				break;
+
+			case DShot_Stream1:
+				dma_int_streamx_mask = DMA_INT_STREAM1_MASK;
+				rSxNDTR = &rS1NDTR(timer);
+				rSxCR = &rS1CR(timer);
+				break;
+
+			case DShot_Stream2:
+				dma_int_streamx_mask = DMA_INT_STREAM2_MASK;
+				rSxNDTR = &rS2NDTR(timer);
+				rSxCR = &rS2CR(timer);
+				break;
+
+			case DShot_Stream3:
+				dma_int_streamx_mask = DMA_INT_STREAM3_MASK;
+				rSxNDTR = &rS3NDTR(timer);
+				rSxCR = &rS3CR(timer);
+				break;
+
+			case DShot_Stream4:
+				dma_int_streamx_mask = DMA_INT_STREAM4_MASK;
+				rSxNDTR = &rS4NDTR(timer);
+				rSxCR = &rS4CR(timer);
+				break;
+
+			case DShot_Stream5:
+				dma_int_streamx_mask = DMA_INT_STREAM5_MASK;
+				rSxNDTR = &rS5NDTR(timer);
+				rSxCR = &rS5CR(timer);
+				break;
+
+			case DShot_Stream6:
+				dma_int_streamx_mask = DMA_INT_STREAM6_MASK;
+				rSxNDTR = &rS6NDTR(timer);
+				rSxCR = &rS6CR(timer);
+				break;
+
+			case DShot_Stream7:
+			default:
+				dma_int_streamx_mask = DMA_INT_STREAM7_MASK;
+				rSxNDTR = &rS7NDTR(timer);
+				rSxCR = &rS7CR(timer);
+				break;
+			}
+
+			// Check if DMA is still in progress (just to be safe, not expected to happen)
+			if (*rSxCR & DMA_SCR_EN) {
+				continue;
+			}
+
+			uint8_t motors_number = dshot_handler[timer].motors_number;
+			dshot_dmar_data_prepare(timer, first_motor, motors_number);
+
+			// Flush cache so DMA sees the data
+			up_clean_dcache((uintptr_t)dshot_burst_buffer[timer],
+					(uintptr_t)dshot_burst_buffer[timer] + DSHOT_BURST_BUFFER_SIZE(motors_number));
+
+			first_motor += motors_number;
+
+			uint32_t dshot_data_size = motors_number * ONE_MOTOR_BUFF_SIZE;
+			rLIFCR(timer) |= dma_int_streamx_mask; //clear interrupt flags
```

Comment:
From stream4 to stream7  rHIFCR should be cleared.
![image](https://user-images.githubusercontent.com/10188706/65550135-33599580-df1f-11e9-9e47-d52bb15b4d49.png)

Reply:
Thanks, updated.